src/Microsoft.Diagnostics.Monitoring.EventPipe/DistributedTraces/DistributedTracesPipeline.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tracing;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe
+{
+    internal class DistributedTracesPipeline : EventSourcePipeline<DistributedTracesPipelineSettings>
+    {
+        private readonly IActivityLogger[] _loggers;
+
+        public DistributedTracesPipeline(DiagnosticsClient client,
+            DistributedTracesPipelineSettings settings,
+            IEnumerable<IActivityLogger> loggers) : base(client, settings)
+        {
+            _loggers = loggers?.ToArray() ?? throw new ArgumentNullException(nameof(loggers));
+        }
+
+        protected override MonitoringSourceConfiguration CreateConfiguration()
+            => new ActivitySourceConfiguration(Settings.SamplingRatio, Settings.Sources);
+
+        protected override async Task OnRun(CancellationToken token)
+        {
+            double samplingRatio = Settings.SamplingRatio;
+            if (samplingRatio < 1D)
+            {
+                await ValidateEventSourceVersion().ConfigureAwait(false);
+            }
+
+            await base.OnRun(token).ConfigureAwait(false);
+        }
+
+        private async Task ValidateEventSourceVersion()
+        {
+            int majorVersion = 0;
+
+            using CancellationTokenSource cancellationTokenSource = new();
+
+            DiagnosticsEventPipeProcessor processor = new(
```

Comment:
you should put await using in front of processor to ensure that it gets disposed even during unexpected exception throws